### ISSUE TITLE
A fit() or similar function in training part.

### ISSUE THREAD

**Issue by beistvan** [beistvan/leaffliction#11]
Check to see if there's a `fit()` or other train function we need in training, or if it's correct as it is now, or if fixes are needed for training.
Probably in the `main()` function of `src/model/train.py` after model creation.

`...
    model = create_model(num_classes)
    model.summary()

    val_loss, val_acc = model.evaluate(val_generator)
    print(f"Validation accuracy: {val_acc:.2f}")

    model.save("trained_leaf_disease_model.h5")
    print("Model saved as trained_leaf_disease_model.h5")
...`

**Comment by beistvan** [comment]
Function fit() is added to the `training.py` after `model.summary()`
...
```python
    model.summary()
    model.fit(
        train_generator,
        validation_data=val_generator,
        batch_size=args.batch_size,
        epochs=args.epochs,
        verbose=1
    )
```
...
